From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sairamreddy Bojja <sairamreddy.bojja@oss.qualcomm.com>
Date: Tue, 11 Aug 2026 22:14:59 +0530
Subject: [PATCH] arm64: dts: qcom: glymur: Fix missing SMMU stream IDs for
 ADSP

Add the missing SIDs for the LPASS eNPU masters to the ADSP
iommus list. These masters are exercised during audio record
use cases where Environmental Acoustic Intelligence (EAI)
models run on the eNPUs.

Currently all transactions from these eNPU masters will result
in global faults; fix them by adding the additional SIDs.

Fixes: cd1d174c75f7 ("arm64: dts: qcom: glymur: Add ADSP and CDSP for Glymur SoC")
Signed-off-by: Sairamreddy Bojja <sairamreddy.bojja@oss.qualcomm.com>
Co-developed-by: Iranna Mundaganur <iranna.mundaganur@oss.qualcomm.com>
Signed-off-by: Iranna Mundaganur <iranna.mundaganur@oss.qualcomm.com>
Reviewed-by: Abel Vesa <abel.vesa@oss.qualcomm.com>
Link: https://lore.kernel.org/r/20260811-adsp-smmu-fix-v1-1-a596d284c455@oss.qualcomm.com
---
 arch/arm64/boot/dts/qcom/glymur.dtsi | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/glymur.dtsi b/arch/arm64/boot/dts/qcom/glymur.dtsi
index 30d96bc2e16..209ba917c0b 100644
--- a/arch/arm64/boot/dts/qcom/glymur.dtsi
+++ b/arch/arm64/boot/dts/qcom/glymur.dtsi
@@ -4100,7 +4100,9 @@ remoteproc_adsp: remoteproc@6800000 {
 			compatible = "qcom,glymur-adsp-pas", "qcom,sm8550-adsp-pas";
 			reg = <0x0 0x06800000 0x0 0x10000>;
 
-			iommus = <&apps_smmu 0x1000 0x0>;
+			iommus = <&apps_smmu 0x1000 0x0>,
+				 <&apps_smmu 0x1040 0x20>,
+				 <&apps_smmu 0x1080 0x0>;
 
 			interrupts-extended = <&pdc 6 IRQ_TYPE_EDGE_RISING>,
 					      <&smp2p_adsp_in 0 IRQ_TYPE_EDGE_RISING>,
-- 
2.56.0

